polkadot/primitives/src/v6/mod.rs

```diff
@@ -735,6 +759,37 @@ impl<H> BackedCandidate<H> {
 	{
 		self.candidate.to_plain()
 	}
+
+	/// Get a copy of the validator indices and the assumed core index, if any.
+	pub fn validator_indices_and_core_index(
+		&self,
+		core_index_enabled: bool,
+	) -> (&BitSlice<u8, bitvec::order::Lsb0>, Option<CoreIndex>) {
+		// This flag tells us if the block producers must enable Elastic Scaling MVP hack.
+		// It extends `BackedCandidate::validity_indices` to store a 8 bit core index.
+		if core_index_enabled {
+			let core_idx_offset = self.validator_indices.len().saturating_sub(8);
```

Comment:
hmm, how is this related to core count? I think it would be a problem to assume that the core index is present if the length is greater than 8 for a backing group size larger than 8. Isn't that right?

Is it fair to assume that backing group sizes won't grow past 8? If it is, maybe we should add some runtime logic that forbids this

Reply:
LOL. I was completely off here and confused two things. Right the limit is the backing group size - which we are more on the edge of reducing in size than increasing. Maybe someone has a testnetwork with large backing groups :thinking: Indeed adding a check to configuration that groups can't be bigger than 8 might make sense. Ok so options are:

1. Leave as is.
2. Do limit group sizes to <=8.
3. Add additional bits which indicate how the rest should be treated.

At this point I am also leaning towards (1). It is ugly, but it is also a hacky interim solution ... so, might as well look like it.